From 55eb8c95a89f32aec16b7764e78e8cf75169dc81 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sat, 17 Feb 2018 06:26:10 -0800
Subject: [PATCH] ld: add soname to shareable objects

so that OE's shared library dependency handling
can find them.

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Matt Madison <matt@madison.systems>

---
 src/cmd/link/internal/ld/lib.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/cmd/link/internal/ld/lib.go b/src/cmd/link/internal/ld/lib.go
index 220aab3..703925f 100644
--- a/src/cmd/link/internal/ld/lib.go
+++ b/src/cmd/link/internal/ld/lib.go
@@ -1135,6 +1135,7 @@ func (ctxt *Link) hostlink() {
				argv = append(argv, "-Wl,-z,relro")
			}
			argv = append(argv, "-shared")
+			argv = append(argv, fmt.Sprintf("-Wl,-soname,%s", filepath.Base(*flagOutfile)))
			if ctxt.HeadType != objabi.Hwindows {
				// Pass -z nodelete to mark the shared library as
				// non-closeable: a dlclose will do nothing.
@@ -1146,6 +1147,8 @@ func (ctxt *Link) hostlink() {
			argv = append(argv, "-Wl,-z,relro")
		}
		argv = append(argv, "-shared")
+		argv = append(argv, fmt.Sprintf("-Wl,-soname,%s", filepath.Base(*flagOutfile)))
+
	case BuildModePlugin:
		if ctxt.HeadType == objabi.Hdarwin {
			argv = append(argv, "-dynamiclib")
@@ -1154,6 +1157,7 @@ func (ctxt *Link) hostlink() {
				argv = append(argv, "-Wl,-z,relro")
			}
			argv = append(argv, "-shared")
+			argv = append(argv, fmt.Sprintf("-Wl,-soname,%s", filepath.Base(*flagOutfile)))
		}
	}
